src/topologies/halfedge.jl

```diff
@@ -338,3 +328,68 @@ nfacets(t::HalfEdgeTopology) = length(t.halfedges) ÷ 2
 # ------------
 
 Base.convert(::Type{HalfEdgeTopology}, t::Topology) = HalfEdgeTopology(collect(elements(t)))
+
+# -----------------
+# HELPER FUNCTIONS
+# -----------------
+
+function adjsort(elems::AbstractVector{<:Connectivity})
+  # initialize list of adjacent elements
+  # with first element from original list
+  list = indices.(elems)
+  adjs = Tuple[popfirst!(list)]
+
+  # the loop will terminate if the mesh
+  # is manifold, and that is always true
+  # with half-edge topology
+  while !isempty(list)
+    # lookup all elements that share at least
+    # one vertex with the last adjacent element
+    found = false
+    vinds = last(adjs)
+    for i in vinds
+      einds = findall(e -> i ∈ e, list)
+      if !isempty(einds)
+        # lookup all elements that share at
+        # least two vertices (i.e. edge)
+        for j in sort(einds, rev=true)
+          if length(vinds ∩ list[j]) > 1
+            found = true
+            push!(adjs, popat!(list, j))
+          end
+        end
+      end
+    end
+
+    if !found && !isempty(list)
+      # we are done with this connected component
+      # pop a new element from the original list
+      push!(adjs, popfirst!(list))
+    end
+  end
+
+  connect.(adjs)
+end
+
+function anyhalf(inds, half4pair)
+  n = length(inds)
+  for i in eachindex(inds)
+    uv = (inds[i], inds[mod1(i + 1, n)])
+    if haskey(half4pair, uv)
+      return true
+    end
+  end
+  return false
+end
+
+function anyhalfclaimed(inds, half4pair)
+  ∅ = HalfEdge(0, nothing)
```

Comment:
This can't be defined internally; c.f. the bug in base Julia I discovered, discussed here https://discourse.julialang.org/t/unmutated-mutable-type-being-corrupted/129161 and at JuliaLang/julia#58470

Reply:
Wow! I am starting to feel concerned about Julia >= v1.11. Can we rewrite this function to avoid the `∅` variable? Let's be more explicit with an additional `if` statement to avoid the bug in present and future versions of the language.

Reply:
The change I pushed in [7f0dab5](https://github.com/JuliaGeometry/Meshes.jl/pull/1194/commits/7f0dab5bce4a1c65ccfe7178e358484725193537) successfully avoids the issue from what I can tell/test.